pkg/virt-handler/container-disk/mount_test.go

```diff
@@ -174,4 +179,226 @@ var _ = Describe("ContainerDisk", func() {
 			Expect(record).To(BeNil())
 		})
 	})
+
+	Context("containerdisks checksum", func() {
+		var rootMountPoint string
+
+		diskContent := []byte{0x6B, 0x75, 0x62, 0x65, 0x76, 0x69, 0x72, 0x74}
+
+		BeforeEach(func() {
+			ctrl := gomock.NewController(GinkgoT())
+			mockIsolationDetector := isolation.NewMockPodIsolationDetector(ctrl)
+			mockNodeIsolationResult := isolation.NewMockIsolationResult(ctrl)
+			mockPodIsolationResult := isolation.NewMockIsolationResult(ctrl)
+
+			m.podIsolationDetector = mockIsolationDetector
+			m.nodeIsolationResult = mockNodeIsolationResult
+
+			m.socketPathGetter = func(vmi *v1.VirtualMachineInstance, volumeIndex int) (string, error) {
+				return "somewhere", nil
+			}
+			m.kernelBootSocketPathGetter = func(vmi *v1.VirtualMachineInstance) (string, error) {
+				return "somewhere-kernel", nil
+			}
+
+			mockIsolationDetector.EXPECT().DetectForSocket(vmi, gomock.Any()).Return(mockPodIsolationResult, nil)
+
+			mockPodIsolationResult.EXPECT().Mounts(gomock.Any()).Return([]*mount.Info{&mount.Info{Root: "/", Mountpoint: "/disks"}}, nil)
+
+			rootMountPoint, err = os.MkdirTemp(tmpDir, "root")
+			Expect(err).ToNot(HaveOccurred())
+			partentToChildMountPoint, err := os.MkdirTemp(rootMountPoint, "child")
+			Expect(err).ToNot(HaveOccurred())
+			mockNodeIsolationResult.EXPECT().Mounts(gomock.Any()).Return([]*mount.Info{&mount.Info{Root: partentToChildMountPoint}}, nil)
+
+			rootMountPointSafePath, err := safepath.NewPathNoFollow(rootMountPoint)
+			Expect(err).ToNot(HaveOccurred())
+			mockNodeIsolationResult.EXPECT().MountRoot().Return(rootMountPointSafePath, nil)
+		})
+
+		Context("verification", func() {
+
+			type args struct {
+				storedChecksum uint32
+				diskContent    []byte
+				shouldSucceed  bool
+				expectedError  error
+			}
+
+			DescribeTable(" should", func(args *args) {
+				vmiVolume := vmi.Spec.Volumes[0]
+
+				diskFile, err := os.Create(filepath.Join(rootMountPoint, vmiVolume.ContainerDisk.Path))
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = (diskFile.Write(args.diskContent))
+				Expect(err).ToNot(HaveOccurred())
+				defer diskFile.Close()
```

Comment:
This can be simplified:
```suggestion
				err := os.WriteFile(filepath.Join(rootMountPoint, vmiVolume.ContainerDisk.Path), args.diskContent, 0660)
				Expect(err).ToNot(HaveOccurred())
```

Reply:
sure, will change.